drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -280,12 +299,71 @@ def studio_submit(self, submissions, suffix=''):
         self.weight = float(submissions['weight'])
         self.item_background_color = submissions['item_background_color']
         self.item_text_color = submissions['item_text_color']
+        # Possible ValueError should be catched in _validate_submissions
+        self.max_items_per_zone = self._get_max_items_per_zone(submissions)
         self.data = submissions['data']
 
         return {
             'result': 'success',
         }
 
+    @staticmethod
+    def _get_max_items_per_zone(submissions):
+        """
+        Parses Max items per zone value coming from editor.
+
+        Does not catch ValueError intentionally to allow validation to capture this problem
+        """
+        raw_max_items_per_zone = submissions.get('max_items_per_zone', None)
+        if not raw_max_items_per_zone:
+            return None
+        else:
+            return int(raw_max_items_per_zone)
+
+    def _validate_submissions(self, submissions):
+        """
+        Validates that submissions passed are correct
+        """
+        errors = []
+
+        items = submissions['data']['items']
+
+        try:
+            max_items_per_zone = self._get_max_items_per_zone(submissions)
+        except ValueError as exc:
+            errors.append(_('Failed to parse "Max items per zone" - please check settings'))
+            logger.exception(exc)
+            return errors
+
+        if max_items_per_zone:
+            if max_items_per_zone < 0:
+                errors.append(_('"Max items per zone" should be positive, got {max_items_per_zone}').format(
+                    max_items_per_zone=max_items_per_zone
+                ))
+            else:
+                errors.extend(self._validate_item_zones(items, max_items_per_zone))
+
+        return errors
+
+    @staticmethod
+    def _validate_item_zones(items, max_items_per_zone):
+        errors = []
+        counter = Counter()
+        for item in items:
+            for zone in item.get('zones', []):
```

Comment:
@itsjeyd I don't think an old DnDv2 still using `item['zone']` will have this validation code - not like we're separately deploying backend and frontend here, so when this validation is added, JS will use `item["zones"]` - [it is already in master](https://github.com/edx-solutions/xblock-drag-and-drop-v2/blob/630a8d7f519409c6c7237660557823998d9e9c29/drag_and_drop_v2/public/js/drag_and_drop_edit.js#L485-L487), while this validation is not.